code/build/template/include/extensions/[+python orxPy.h]

```diff
@@ -0,0 +1,2613 @@
+
+    BIND_METHOD("set_volume(self, volume: float) -> None", set_volume);
+    BIND_METHOD("set_pitch(self, pitch: float) -> None", set_pitch);
+    BIND_METHOD("set_panning(self, panning: float, mix: bool) -> None", set_panning);
+
+    BIND_METHOD("play(self) -> None", play);
+    BIND_METHOD("stop(self) -> None", stop);
+
+    BIND_METHOD("add_filter(self, name: str) -> None", add_filter);
+    BIND_METHOD("remove_last_filter(self) -> None", remove_last_filter);
+    BIND_METHOD("remove_all_filters(self) -> None", remove_all_filters);
+
+    BIND_METHOD("add_shader(self, name: str, recursive: bool = False) -> None", add_shader);
+    BIND_METHOD("remove_shader(self, name: str, recursive: bool = False) -> None", remove_shader);
+    BIND_METHOD("enable_shader(self, enabled: bool = True) -> None", enable_shader);
+    BIND_METHOD("is_shader_enabled(self) -> bool", is_shader_enabled);
+
+    BIND_METHOD("add_time_line_track(self, name: str, recusive: bool = False) -> None", add_time_line_track);
+    BIND_METHOD("remove_time_line_track(self, name: str, recursive: bool = False) -> None", remove_time_line_track);
+    BIND_METHOD("enable_time_line(self, enabled: bool = True) -> None", enable_time_line);
+    BIND_METHOD("is_time_line_enabled(self) -> bool", is_time_line_enabled);
+
+    BIND_METHOD("add_trigger(self, name: str, recursive: bool = False) -> None", add_trigger);
+    BIND_METHOD("remove_trigger(self, name: str, recursive: bool = False) -> None", remove_trigger);
+    BIND_METHOD("fire_trigger(self, name: str, refinement: list[str] | None = None, recursive: bool = False) -> None", fire_trigger);
+
+    BIND_METHOD("get_name(self) -> str", get_name);
+
+    BIND_METHOD("set_rgb(self, rgb: Vector, recursive: bool = False) -> None", set_rgb);
+    BIND_METHOD("get_rgb(self) -> Vector", get_rgb);
+
+    BIND_METHOD("set_alpha(self, alpha: float, recursive: bool = False) -> None", set_alpha);
+    BIND_METHOD("get_alpha(self) -> float", get_alpha);
+
+    BIND_METHOD("set_life_time(self, life_time: float | str | None) -> None", set_life_time);
+    BIND_METHOD("get_life_time(self) -> float | None", get_life_time);
+
+    BIND_METHOD("get_active_time(self) -> float", get_active_time);
+    BIND_METHOD("reset_active_time(self, recursive: bool = False) -> None", reset_active_time);
+
+#undef BIND_METHOD
+
+    // Bind object functions
+    py_bind(pyMod, "create_from_config(section_name: str) -> Object | None", BINDING_NAME(Object, __new__));
+    py_bind(pyMod, "from_guid(guid: int) -> Object | None", BINDING_NAME(Object, __new__));
+    py_bind(pyMod, "raycast(begin: Vector, end: Vector, self_flags: int, check_mask: int, early_exit: bool = False) -> tuple[Object, Vector, Vector] | None", BINDING_NAME(Object, raycast));
+  }
+
+  BIND(Orx, log)
+  {
+    PY_CHECK_ARGC(1);
+    ARG_STR(zMessage, 0);
+    orxLOG("%s", zMessage);
+    RETURN_NONE;
+  }
+
+  BIND(Orx, close)

```

Comment:
This is a bit of an ad-hoc definition, but the functionality is very nice to have and is used in non-Python fresh `init` projects.